scripts/python/lib/config.py

```diff
@@ -1871,19 +2000,58 @@ def yield_ntmpl_phyintf_data_switch(self, node_template_index):
             yield member
 
     def get_ntmpl_phyintf_data_dev(
-            self, node_template_index, index=None):
+            self, node_template_index, index):
         """Get node_templates physical_interfaces data dev
         Args:
             node_template_index (int): Node template index
-            index (int, optional): List index
+            index (int): List index
 
         Returns:
-            str or list of str: Data dev member or list
+            str: Data interface device (iface) value
         """
 
         node_template = self.cfg.node_templates[node_template_index]
-        return self._get_members(
-            node_template.physical_interfaces.data, self.CfgKey.DEVICE, index)
+        if_label = node_template.physical_interfaces.data[index].interface
+        return self.lookup_interface_iface(if_label)
+
+    def get_ntmpl_phyintf_pxe_dev(
+            self, node_template_index, index=0):
+        """Get node_templates physical_interfaces PXE dev
+        Args:
+            node_template_index (int): Node template index
+            index (int, optional): List index (defaults to 0)
+
+        Returns:
+            str: PXE interface device (iface) value
+        """
+
+        node_template = self.cfg.node_templates[node_template_index]
+        if_label = node_template.physical_interfaces.pxe[index].interface
+        return self.lookup_interface_iface(if_label)
+
+    def lookup_interface_iface(self, if_label):
+        """Get node_templates physical_interfaces data dev
```

Comment:
'Get node_templates physical_interfaces data dev' should probably say get interface template data device'

Reply:
Agreed.